client/src/modules/core/webpackmodules.js

```
@@ -7,85 +7,210 @@
+    SoundModule: Filters.byProperties(["playSound"]),
+
+    /* Window, DOM, HTML */
+    WindowInfo: Filters.byProperties(['isFocused', 'windowSize']),
+    TagInfo: Filters.byProperties(['VALID_TAG_NAMES']),
+    DOMInfo: Filters.byProperties(['canUseDOM']),
+    HTMLUtils: Filters.byProperties(['htmlFor', 'sanitizeUrl']),
+
+    /* Locale/Location and Time */
+    LocaleManager: Filters.byProperties(['setLocale']),
+    Moment: Filters.byProperties(['parseZone']),
+    LocationManager: Filters.byProperties(["createLocation"]),
+    Timestamps: Filters.byProperties(["fromTimestamp"]),
+
+    /* Strings and Utils */
+    Strings: Filters.byProperties(["TEXT", "TEXTAREA_PLACEHOLDER"]),
+    StringFormats: Filters.byProperties(['a', 'z']),
+    StringUtils: Filters.byProperties(["toASCII"]),
+
+    /* URLs and Utils */
+    URLParser: Filters.byProperties(['Url', 'parse']),
+    ExtraURLs: Filters.byProperties(['getArticleURL']),
+
+
+    /* DOM/React Components */
+    /* ==================== */
+    UserSettingsWindow: Filters.byProperties(['open', 'updateAccount']),
+    LayerManager: Filters.byProperties(['popLayer', 'pushLayer']),
+
+    /* Modals */
+    ModalStack: Filters.byProperties(['push', 'update', 'pop', 'popWithKey']),
+    UserProfileModals: Filters.byProperties(['fetchMutualFriends', 'setSection']),
+    ConfirmModal: Filters.byPrototypeFields(['handleCancel', 'handleSubmit', 'handleMinorConfirm']),
+
+    /* Popouts */
+    PopoutStack: Filters.byProperties(['open', 'close', 'closeAll']),
+    PopoutOpener: Filters.byProperties(['openPopout']),
+    EmojiPicker: Filters.byPrototypeFields(['onHoverEmoji', 'selectEmoji']),
+
+    /* Context Menus */
+    ContextMenuActions: Filters.byCode(/CONTEXT_MENU_CLOSE/, c => c.close),
+    ContextMenuItemsGroup: Filters.byCode(/itemGroup/),
+    ContextMenuItem: Filters.byCode(/\.label\b.*\.hint\b.*\.action\b/),
+
+    /* In-Message Links */
+    ExternalLink: Filters.byCode(/\.trusted\b/)
```

I like the idea of such list, but keeping unused stuff is dangerous. We definitely need to develop some kind of integration tests, to know when something from this list stops working on canary.

---

Yes we will definitely need some tests and such to see when things stop working. But I believe plugins that want to rely on it should check if the return is null so they don't error out

---

BTW, it might be the case of async versions. They might reject instead of resolve with null if there is no such module. But this isn't the case in current implementation. Btw look at new comment below